plugins/@grouparoo/google-sheets/src/lib/sheet-export/exportRecords.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  BatchConfig,
+  BatchGroupMode,
+  BatchMethodAddToGroups,
+  BatchMethodCreateByForeignKeyAndSetDestinationIds,
+  BatchMethodDeleteByDestinationIds,
+  BatchMethodFindAndSetDestinationIds,
+  BatchMethodGetClient,
+  BatchMethodNormalizeForeignKeyValue,
+  BatchMethodNormalizeGroupName,
+  BatchMethodRemoveFromGroups,
+  BatchMethodUpdateByDestinationIds,
+  BatchSyncMode,
+  buildBatchExports,
+  exportRecordsInBatch,
+} from "@grouparoo/app-templates/dist/destination/batch";
+import { ExportRecordsPluginMethod } from "@grouparoo/core";
+import Spreadsheet from "../shared/Spreadsheet";
+
+// return an object that you can connect with
+const getClient: BatchMethodGetClient = async ({ config }) => {
+  return new Spreadsheet(
+    config.appOptions,
+    config.destinationOptions.sheet_url?.toString()
+  );
+};
+
+// fetch using the keys to set destinationId and result on BatchExports
+// use the getByForeignKey to lookup results
+const findAndSetDestinationIds: BatchMethodFindAndSetDestinationIds = async ({
+  client,
+  foreignKeys,
+  getByForeignKey,
+  config,
+}) => {
+  for (const foreignKey of foreignKeys) {
+    const user = getByForeignKey(foreignKey);
+    try {
+      const row = await client.getRowByPrimaryKey(
+        config.foreignKey,
+        foreignKey
+      );
+      if (row) {
+        user.destinationId = row._rowNumber;
```

Comment:
I think this is what I'm most worried about here, though I'm not sure of a better approach. If this row number changes because someone (or grouparoo) inserts/deletes, things are going to get crazy.